src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private double randomNumber;
+
+    public Figure getRandomFigure() {
+        getRandomFigureNumber();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        if (randomNumber == 0) {
+            return new Circle("RandomCircle",
+                    colorSupplier.getRandomColor().toString(),
+                    getRandomSide());
+        } else if (randomNumber == 1) {
+            return new Square("RandomSquare",
+                    colorSupplier.getRandomColor().toString(),
+                    getRandomSide());
+        } else if (randomNumber == 2) {
+            return new Rectangle("RandomRectangle",
+                    getDefaultFigure().getColor().toString(),
+                    getRandomSide(),
+                    getRandomSide());
+        } else if (randomNumber == 3) {
+            return new IsoscelesTrapezoid("RandomIsoscelesTrapezoid",
+                    getDefaultFigure().getColor().toString(),
+                    getRandomSide(),
+                    getRandomSide(),
+                    getRandomSide());
+        } else {
+            return new RightTriangle("RandomRightTriangle",
+                    getDefaultFigure().getColor().toString(),
+                    getRandomSide(),
+                    getRandomSide());
+        }
+    }
+
+    //get random number for chose random what figure create
+    public void getRandomFigureNumber() {
+        this.randomNumber = random.nextInt(5);
+    }
+
+    //get random number for create random figure parameters
+    public double getRandomSide() {
+        return randomNumber = random.nextInt(10) + 1;
+    }
+
+    //create default figure
+    public Figure getDefaultFigure() {
+        return new Circle("defaultFigure", "WHITE", 10);
```

Comment:
All magic numbers, strings, and symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
Use `Enum` instead of `String`